examples/src/main/java/org/apache/spark/examples/ml/JavaGaussianMixtureExample.java

```diff
@@ -0,0 +1,108 @@
+ * limitations under the License.
+ */
+
+package org.apache.spark.examples.ml;
+
+import java.util.regex.Pattern;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.SQLContext;
+import org.apache.spark.sql.catalyst.expressions.GenericRow;
+// $example on$
+import org.apache.spark.ml.clustering.GaussianMixture;
+import org.apache.spark.ml.clustering.GaussianMixtureModel;
+import org.apache.spark.ml.clustering.GaussianMixtureSummary;
+import org.apache.spark.mllib.linalg.Vector;
+import org.apache.spark.mllib.linalg.VectorUDT;
+import org.apache.spark.mllib.linalg.Vectors;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+// $example off$
+
+
+/**
+ * An example demonstrating a Gaussian Mixture Model.
+ * Run with
+ * <pre>
+ * bin/run-example ml.JavaGaussianMixtureExample <file> <k>
+ * </pre>
+ */
+public class JavaGaussianMixtureExample {
+
+  private static class ParsePoint implements Function<String, Row> {
+    private static final Pattern separator = Pattern.compile(" ");
+
+    @Override
+    public Row call(String line) {
+      String[] tok = separator.split(line.trim());
+      double[] point = new double[tok.length];
+      for (int i = 0; i < tok.length; ++i) {
+        point[i] = Double.parseDouble(tok[i]);
+      }
+      Vector[] points = {Vectors.dense(point)};
+      return new GenericRow(points);
+    }
+  }
+
+  public static void main(String[] args) {
+    if (args.length != 2) {
+      System.err.println("Usage: ml.JavaGaussianMixtureExample <file> <k>");
+      System.exit(1);
+    }
+    String inputFile = args[0];
+    int k = Integer.parseInt(args[1]);
+
+    // Parses the arguments
+    SparkConf conf = new SparkConf().setAppName("JavaGaussianMixtureExample");
+    JavaSparkContext jsc = new JavaSparkContext(conf);
+    SQLContext sqlContext = new SQLContext(jsc);
+
+    // $example on$
+    // Loads data
+    JavaRDD<Row> points = jsc.textFile(inputFile).map(new ParsePoint());
+    StructField[] fields = {new StructField("features", new VectorUDT(), false, Metadata.empty())};
+    StructType schema = new StructType(fields);
+    Dataset<Row> dataset = sqlContext.createDataFrame(points, schema);
+
+    // Trains a GaussianMixture model
+    GaussianMixture gmm = new GaussianMixture()
+      .setK(k);
+    GaussianMixtureModel model = gmm.fit(dataset);
+
+    GaussianMixtureSummary summary = model.summary();
+
+    // Show cluster centers of the transformed data.
+    summary.cluster().show();
+
+    // Show probability of each cluster.
+    summary.probability().show();
+
+    // Print size of (number of data points in) each cluster.
+    for (Long count: summary.clusterSizes()) {
```

Comment:
This just prints out a couple of numbers when it is run, with no explanation of what they are. I think this should match the scala output as closely as possible. You can change it when we decide what the exact output should be.